src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -533,22 +526,65 @@ def test_config_directory_is_file(self):
         with confdir.open("w") as f:
             f.write("dummy\n")
 
-        outcome = yield cli(confdir, ["list"])
+        outcome = yield cli(["--config", confdir.path, "list"])
         self.assertThat(outcome.code, Equals(1))
         self.assertThat(outcome.stderr, Contains("Unable to load configuration"))
 
-    @defer.inlineCallbacks
+    @inline_callbacks
     def test_config_directory_empty(self):
         """
         A directory that is empty isn't valid for --config
         """
         confdir = FilePath(self.mktemp())
         confdir.makedirs()
 
-        outcome = yield cli(confdir, ["list"])
+        outcome = yield cli(["--config", confdir.path, "list"])
         self.assertThat(outcome.code, Equals(1))
         self.assertThat(outcome.stderr, Contains("Unable to load configuration"))
 
+    @inline_callbacks
+    def test_config_directory(self):
+        """
+        Passing --config option loads the configuration from the provided directory.
+        """
+        confdir = FilePath(self.mktemp())
+        nodedir = self.useFixture(
+            NodeDirectory(FilePath(self.mktemp()))
+        )
+        yield magic_folder_initialize(confdir, u"tcp:5555", nodedir.path, None)
+
+        options = magic_folder_cli.MagicFolderCommand()
+        options.parseOptions(["--config", confdir.path, "list"])
+        self.assertThat(
+            options.config,
+            MatchesStructure(
+                basedir=Equals(confdir),
+                api_endpoint=Equals(u"tcp:5555"),
+                tahoe_node_directory=Equals(nodedir.path),
+            )
+        )
+
+    @inline_callbacks
+    def test_default_config_directory(self):
+        """
+        Not passing a --config loads the configuration from the default directory.
+        """
```

Comment:
This test doesn't seem to match what it says here; how is this testing the default config directory?

Reply:
I've added a comment explaining how we are testing the default.